File: pdsa/java/graph/LinkedList.java
package pdsa.graph;

import java.util.Iterator;

public class LinkedList<T> implements Iterable<T> {
    private Node head;

    public void addFirst(T data) {
        if (head == null) {
            head = new Node(data, null);
        } else {
            head = new Node(data, head);
        }
    }

    @Override
    public Iterator iterator() {
        return new LinkedListIterator(head);
    }
}

File: pdsa/csharp/LinkedList.cs
using System;

namespace algorithms
{
	public class LinkedList
	{
		Node head;
		public LinkedList ()
		{
			head = null;
		}

		public void add(int data)
		{
			if (head == null) {
				head = new Node (data, null);
			} else {
				Node current = head;
				while (current.Next != null) {
					current = current.Next;
				}

				current.Next = new Node (data, null);
			}
		}

		public void print()
		{
			Node current = head;
			while (current != null) {
				Console.WriteLine(current.Data);
				current = current.Next;

			}
		}
	}
}

